aggregator_core/src/datastore/test_util.rs

```diff
@@ -120,10 +122,47 @@ impl EphemeralDatabase {
         });
 
         let port_number = port_rx.await.unwrap();
+
+        // Make a best-effort attempt to delete databases created for datastores, once they are no
+        // longer needed. This may fail if the container hosting the database server is deleted
+        // faster, which would make this moot.
+        let (cleanup_tx, cleanup_rx) = mpsc::channel::<String>();
+        std::thread::spawn(move || {
+            let runtime = tokio::runtime::Builder::new_multi_thread()
+                .worker_threads(1)
+                .enable_all()
+                .build()
+                .unwrap();
+            let handle = runtime.handle();
+
+            let connection_string =
+                format!("postgres://postgres:postgres@127.0.0.1:{port_number}/postgres");
+
+            while let Ok(name) = cleanup_rx.recv() {
+                let connection_string = connection_string.clone();
+                handle.spawn({
```

Comment:
optional: I would have connected to the DB once, outside of the recv loop, and kept only the `DROP DATABASE` inside the recv loop without spawning a separate task for each received DB name. We might end up buffering a few database names if dropping a database takes a while, but we'll already effectively do that in the tasks we generate if connecting takes a while. This would more effectively control concurrency, at the cost of the DB cleanup work sharing fate if e.g. the cleanup connection goes bad.

This doesn't need to be addressed (unless it nerd-snipes you) -- I don't want to take too much time on this cleanup, and I don't think a few extra tasks being generated during tests are going to meaningfully harm performance/resource usage.

Reply:
I'd prefer to keep it this way, so we can pipeline `DROP DATABASE` commands inside the database. I tried dropping one database manually, while the container was loaded with other tests, and it took a few seconds. Unit tests can create several databases per second, and a simulation test loop can create a database about once per second. My concern is that serializing `DROP DATABASE` statements would lead to long-running simulation test runs falling behind in cleanup, and then hitting a resource limit. This can be painful, because a whole run can take a half hour, especially if a failing input is found and needs to be shrunk, and the database container becomes unusable by subsequent iterations.